s3upload.js

```diff
@@ -32,16 +37,19 @@ function S3Upload(options) {
             this[option] = options[option];
         }
     }
-    var files = this.fileElement ? this.fileElement.files : this.files || [];
-    this.handleFileSelect(files);
+    this.handleFileSelect(this.files);
```

Comment:
Oh, I see now.  Yeah, this is better.  But I'd worry that someone is using `fileElement` already, and I wouldn't want to break their code.  Let's keep your change in the React component, but revert this, so that we stay compatible.